### PR TITLE
Added str_pad optional arguments

### PR DESCRIPTION
**Pull request by urlund**
According to http://php.net/manual/en/function.str-pad.php

### CODE REVIEW

**Comment by schlessera** [comment]
@urlund Before submitting a pull request, could you create an issue first describing the problem you're trying to solve?

See the [Handbook section about pull requests](https://make.wordpress.org/cli/handbook/pull-requests/) for more details about our processes.

**Comment by urlund** [comment]
@schlessera Done :)

**Comment by gitlost** [comment]
Thanks for the PR @urlund.

I think it would be better (simpler at least) to only support `$pad_type`, as `$pad_string` requires allowing for varying length strings, and I don't see much application for it anyway. 

Also could you add phpunit tests please? (See https://github.com/wp-cli/php-cli-tools/blob/master/tests/test-cli.php#L41).

(Just to note: https://github.com/wp-cli/php-cli-tools/pull/122 will have to be merged once it's approved for the tests to pass on Travis.)

**Comment by urlund** [comment]
Done?

File: tests/test-cli.php [hunk 1]
```diff
@@ -44,6 +44,9 @@ function test_encoded_string_pad() {
 		$this->assertEquals( 7, strlen( \cli\Colors::pad( 'óra', 6 ) ) ); // special characters take one byte
 		$this->assertEquals( 9, strlen( \cli\Colors::pad( '日本語', 6 ) ) ); // each character takes two bytes
 		$this->assertEquals( 17, strlen( \cli\Colors::pad( 'עִבְרִית', 6 ) ) ); // process Hebrew vowels
+		$this->assertEquals( 6, strlen( \cli\Colors::pad( 'hello', 6, false, false, STR_PAD_RIGHT ) ) );
+		$this->assertEquals( 7, strlen( \cli\Colors::pad( 'óra', 6, false, false, STR_PAD_LEFT ) ) ); // special characters take one byte
+		$this->assertEquals( 9, strlen( \cli\Colors::pad( '日本語', 6, false, false, STR_PAD_BOTH ) ) ); // each character takes two bytes
 	}
```
Comment (gitlost) [inline tests/test-cli.php]: That's cool. Could you also add some tests to exercise the new functionality, eg
```php
$this->assertSame( 5, strpos( \cli\Colors::pad( 'hello', 7, false, false, STR_PAD_RIGHT ), ' ' ) );
$this->assertSame( 6, strrpos( \cli\Colors::pad( 'hello', 7, false, false, STR_PAD_RIGHT ), ' ' ) );
$this->assertSame( 0, strpos( \cli\Colors::pad( 'hello', 7, false, false, STR_PAD_LEFT ), ' ' ) );
$this->assertSame( 1, strrpos( \cli\Colors::pad( 'hello', 7, false, false, STR_PAD_LEFT ), ' ' ) );
$this->assertSame( 0, strpos( \cli\Colors::pad( 'hello', 7, false, false, STR_PAD_BOTH ), ' ' ) );
$this->assertSame( 6, strrpos( \cli\Colors::pad( 'hello', 7, false, false, STR_PAD_BOTH ), ' ' ) );
$this->assertSame( \cli\Colors::pad( 'hello', 7 ), \cli\Colors::pad( 'hello', 7, false, false, STR_PAD_RIGHT ) );
```

**Comment by urlund** [comment]
@gitlost better? ;)

**Comment by gitlost** [comment]
@urlund Oh yes!

**Comment by danielbachhuber** [comment]
Tagged v0.11.7: https://github.com/wp-cli/php-cli-tools/releases/tag/v0.11.7

**Pull request merged**